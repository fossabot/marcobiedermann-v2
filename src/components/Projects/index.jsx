import classNames from 'classnames';
import PropTypes from 'prop-types';
import React from 'react';
import Project from '../Project';
import styles from './style.module.css';

const Projects = props => {
  const { className, projects, ...otherProps } = props;

  return (
    <ol className={classNames(className, styles.projects)} {...otherProps}>
      {projects.map(project => (
        <li key={project.id}>
          <Project {...project} />
        </li>
      ))}
    </ol>
  );
};

Projects.propTypes = {
  className: PropTypes.string,
  projects: PropTypes.arrayOf(
    PropTypes.shape({
      id: PropTypes.string,
    }),
  ),
};

Projects.defaultProps = {
  className: '',
  projects: [],
};

export default Projects;
