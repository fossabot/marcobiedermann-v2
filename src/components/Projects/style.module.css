.projects {
  display: grid;
  grid-gap: 1em;
  grid-template-columns: repeat(auto-fill, minmax(28em, 1fr));
  list-style: none;
  margin: 0;
  padding: 0;
}
