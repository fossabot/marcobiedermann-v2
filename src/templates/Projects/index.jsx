import { graphql } from 'gatsby';
import PropTypes from 'prop-types';
import React from 'react';
import Grid from '../../components/Grid';
import Layout from '../../components/Layout';
import Projects from '../../components/Projects';
import Section from '../../components/Section';

const ProjectsTemplate = props => {
  const {
    data: { allContentfulProject },
  } = props;

  const { edges: projects } = allContentfulProject;

  return (
    <Layout>
      <Section>
        <Grid>
          <Projects projects={projects.map(project => project.node)} />
        </Grid>
      </Section>
    </Layout>
  );
};

ProjectsTemplate.propTypes = {
  data: PropTypes.shape({
    allContentfulProject: PropTypes.shape({
      edges: PropTypes.arrayOf(
        PropTypes.shape({
          node: PropTypes.shape(),
        }),
      ),
    }),
  }),
};

ProjectsTemplate.defaultProps = {
  data: null,
};

export const pageQuery = graphql`
  query Projects($limit: Int!, $skip: Int!) {
    allContentfulProject(sort: { fields: [createdAt], order: DESC }, limit: $limit, skip: $skip) {
      edges {
        node {
          id
          slug
          title
          thumbnail {
            fluid {
              ...GatsbyContentfulFluid_withWebp_noBase64
            }
          }
        }
      }
    }
  }
`;

export default ProjectsTemplate;
