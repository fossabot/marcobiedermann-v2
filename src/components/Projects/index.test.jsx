import React from 'react';
import renderer from 'react-test-renderer';
import Projects from '.';

describe('Projects component', () => {
  it('should render correctly', () => {
    const tree = renderer.create(<Projects />).toJSON();

    expect(tree).toMatchSnapshot();
  });
});
