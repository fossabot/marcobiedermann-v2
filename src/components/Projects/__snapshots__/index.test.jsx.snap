// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Projects component should render correctly 1`] = `
<ol
  className="projects"
>
  <li>
    <div
      id="123"
    >
      <figure
        className="project__figure"
      >
        <a
          className="project__link"
          href="/projects/undefined"
        >
          <figcaption
            className="project__caption"
          >
            <h2 />
          </figcaption>
        </a>
      </figure>
    </div>
  </li>
</ol>
`;

exports[`Projects component should render empty projects correctly 1`] = `
<ol
  className="projects"
/>
`;
